### DISCUSSION TITLE
How to detect that the WebView2 control is in a closed state

### DISCUSSION THREAD

**Discussion by WildlifeSOS** [MicrosoftEdge/WebView2Feedback#3965] [Q&A]
My .Net 4.8 Winforms application currently uses WebView2 version 1.0.1245.22
Every so often some users using it tell me that forms containing WebView2 controls don't display any content and I'm trying to find out why.
I've added a CoreWebView2 ProcessFailed event handler to see if that gets triggered.
Looking at the help for CoreWebView2ProcessFailedKind I see that it can return BrowserProcessExited and the description of that indicates that 

1) the WebView2 control moves to the Closed state
2) the app has to recreate a new WebView to recover from this failure

Is there any way of detecting that has happened from any of the WebView2 properties?

**Comment by yunate** [comment]
There is a ProcessFailed event may be helpful:
[sample code](https://github.com/MicrosoftEdge/WebView2Samples/blob/main/SampleApps/WebView2WpfBrowser/MainWindow.xaml.cs#:~:text=webView.CoreWebView2.ProcessFailed%20%2B%3D%20WebView_ProcessFailed%3B)
